### ISSUE TITLE
Virtual disk registration not removed from The Virtual Media Manager of VirtualBox in some cases

### ISSUE THREAD

**Issue by nick-korsakov** [hashicorp/packer#7392]
Info:
- Packer version: v1.3.5
- Host platform: Windows 10 x86-64 (1809)
- Debug log: [output](https://gist.github.com/nick-korsakov/5dbdde56d99f6b8df907ed16b2a4711d)
- The example: [template](https://gist.github.com/nick-korsakov/570be9beca380650aa72c4cf4dc6ebc5)
- ISO: [Windows 10](https://www.microsoft.com/en-us/software-download/windows10ISO)

I would like to create VM which is closely repeats physical hardware:
- SATA(0, 0): HDD1
- SATA(1, 0): HDD2
- SATA(2, 0): DVD

Packer creates:
- SATA(0, 0): HDD1
- SATA(1, 0): DVD

I'm changing this to the desired configuration using:
```json
{
    "builders": [
        {
            "type": "virtualbox-iso",
            "...": "...",
            "vboxmanage": [
                [
                    "storageattach", "{{.Name}}",
                    "--storagectl", "SATA Controller",
                    "--port", "1",
                    "--medium", "none"
                ],
                [
                    "storageattach", "{{.Name}}",
                    "--storagectl", "SATA Controller",
                    "--port", "2",
                    "--type", "dvddrive",
                    "--medium", "{{user `iso_url`}}"
                ],

                [
                    "createmedium", "disk",
                    "--filename", "{{pwd}}/output-{{build_name}}/1_hdd2.vdi",
                    "--size", "244198"
                ],
                [
                    "storageattach", "{{.Name}}",
                    "--storagectl", "SATA Controller",
                    "--port", "1",
                    "--type", "hdd",
                    "--medium", "{{pwd}}/output-{{build_name}}/1_hdd2.vdi"
                ]
            ]
        }
    ]
}
```

It works fine until Packer releases resources from VirtualBox. `1_hdd2.vdi` doesn't removed from The Virtual Media Manager (it stays in the list with the warning "Not connected" because file removed physically).

Next Packer build stops with error:
```
==> virtualbox-iso: Error executing command: VBoxManage error: 0%...10%...20%...30%...40%...50%...60%...70%...80%...90%...
==> virtualbox-iso: Progress state: E_INVALIDARG
==> virtualbox-iso: VBoxManage.exe: error: Failed to create medium
==> virtualbox-iso: VBoxManage.exe: error: Cannot register the hard disk 'C:\packer\output-virtualbox-iso\1_hdd2.vdi' {39d5abf0-91b7-42c9-8b71-cc5dbbbaefd9} because a hard disk 'C:\packer\output-virtualbox-iso\1_hdd2.vdi' with UUID {d5c717f2-c86b-460b-a8b4-6b8d708c1233} already exists
==> virtualbox-iso: VBoxManage.exe: error: Details: code E_INVALIDARG (0x80070057), component VirtualBoxWrap, interface IVirtualBox
```

This happens because SATA(1, 0) initially used by DVD then used by HDD2 and DVD moved to SATA(2,0).

**Comment by nick-korsakov** [comment]
Debug [output](https://gist.github.com/nick-korsakov/5dbdde56d99f6b8df907ed16b2a4711d) updated.

P.S.: Debug output was missed due to WSL inability to pass environment variable PACKER_LOG=1 from Linux to Windows when I ran it as $ PACKER_LOG=1 packer.exe build template.json.

**Comment by sylviamoss** [comment]
Hi @nick-korsakov! By reproducing this I realized that it should have been fixed by https://github.com/hashicorp/packer/pull/8483. 
The disk registration wasn't been removed because of a session lock, as a result of that, the second time you ran the build the disk already existed. Now we added a mechanism to retry until it is unlocked ensuring the deletion will be successful. 
This fix will go out soon in the next release! Let me know if you have any questions :D
